sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystTypeConverters.scala

```diff
@@ -159,17 +159,17 @@ object CatalystTypeConverters {
     override def toCatalystImpl(scalaValue: Any): ArrayData = {
       scalaValue match {
         case a: Array[_] =>
-          new GenericArrayData(a.map(elementConverter.toCatalyst))
+          GenericArrayData.allocate(a.map(elementConverter.toCatalyst))
```

Comment:
Yes, I feel so for now since type of `toCatalyst` seems to be `Any`. Is it better to prepare specialized code to make a type concrete here?